popdat/src/import_census.rs

```diff
@@ -1,128 +1,117 @@
-use std::convert::TryFrom;
-
 use geo::algorithm::intersects::Intersects;
-use geojson::GeoJson;
 
 use abstutil::Timer;
 use map_model::Map;
 
 use crate::CensusArea;
 
 impl CensusArea {
-    pub fn find_data_for_map(map: &Map, timer: &mut Timer) -> anyhow::Result<Vec<CensusArea>> {
-        // TODO eventually it'd be nice to lazily download the info needed. For now we expect a
-        // prepared geojson file to exist in data/system/<city>/population_areas.geojson
-        //
-        // expected geojson formatted contents like:
-        // {
-        //     "type": "FeatureCollection",
-        //     "features": [
-        //          {
-        //              "type": "Feature",
-        //              "properties": { "population": 123 },
-        //              "geometry": {
-        //                  "type": "MultiPolygon",
-        //                  "coordinates": [ [ [ [ -73.7, 40.8 ], [ -73.7, 40.8 ], ...] ] ] ]
-        //               }
-        //          },
-        //          {
-        //              "type": "Feature",
-        //              "properties": { "population": 249 },
-        //              "geometry": {
-        //                  "type": "MultiPolygon",
-        //                  "coordinates": [ [ [ [ -73.8, 40.8 ], [ -73.8, 40.8 ], ...] ] ]
-        //               }
-        //           },
-        //          ...
-        //      ]
-        // }
-        //
-        // Note: intuitively you might expect a collection of Polygon's rather than  MultiPolygons,
-        // but anecdotally, the census data I've seen uses MultiPolygons. In practice almost
-        // all are MultiPoly's with just one element, but some truly have multiple polygons.
-        //
-        // When we implement downloading, importer/src/utils.rs has a download() helper that we
-        // could copy here. (And later dedupe, after deciding how this crate will integrate with
-        // the importer)
-        let path = abstutil::path(format!(
-            "system/{}/population_areas.geojson",
-            map.get_name().city
-        ));
-        let bytes = abstutil::slurp_file(&path).map_err(|s| anyhow!(s))?;
-        debug!("parsing geojson at path: {}", &path);
-        timer.start("parsing geojson");
-        let geojson = GeoJson::from_reader(&*bytes)?;
-        timer.stop("parsing geojson");
-        let mut results = vec![];
-        let collection = geojson::FeatureCollection::try_from(geojson)?;
+    pub fn fetch_all_for_map(map: &Map, timer: &mut Timer) -> anyhow::Result<Vec<CensusArea>> {
+        timer.start("processing population areas fgb");
+        let areas = tokio::runtime::Runtime::new()
```

Comment:
It changes from map to map, but anecdotally, this seems to require 8-10 requests totaling 3-4MB of bandwidth.

Reply:
That's _awesome_